src/com/linkedin/parseq/TaskDescriptor.java

```diff
@@ -0,0 +1,20 @@
+package com.linkedin.parseq;
+
+/**
+ * An API to provide description for task.
+ *
+ * parseq-lambda-names provide an implementation for this interface using Java SPI. Any changes made to this interface
+ * might require a change to implementation in parseq-lambda-names.
+ *
+ * @author Siddharth Sodhani (ssodhani@linkedin.com)
+ */
+public interface TaskDescriptor {
```

Comment:
I am not sure how would it work in JAVA SPI where in parseq-lambda-names has to provide implementation for this interface. Moreover I feel keeping as interface it looks more cleaner.